homeassistant/components/filter/sensor.py

```diff
@@ -197,7 +197,15 @@ def _update_filter_sensor_state_event(self, event):
     @callback
     def _update_filter_sensor_state(self, new_state, update_ha=True):
         """Process device state changes."""
-        if new_state is None or new_state.state in [STATE_UNKNOWN, STATE_UNAVAILABLE]:
+        if new_state is None:
+            _LOGGER.warning(
+                "While updating filter %s, the new_state is None.", self._name
```

Comment:
Log messages should not end with a full stop.

```suggestion
                "While updating filter %s, the new_state is None", self._name
```

Reply:
Do we need to throw a warning for this? How could a user solve this warning?

Reply:
I don't understand how the event has a State = None (and not State.state = None) and this is causing several issue reports. 

The warning will help with debugging by users (they will create new issues hopefully) and will eventually go away in the future.

Reply:
These issues are all very recently (starting from 0.118) while the filter sensor has existed unchanged for more then a year :( 

Can't figure out which change has caused these...